core/app/models/spree/wallet_payment_source.rb

```diff
@@ -0,0 +1,17 @@
+class Spree::WalletPaymentSource < ActiveRecord::Base
+  belongs_to :user, class_name: Spree.user_class, foreign_key: 'user_id'
+  belongs_to :payment_source, polymorphic: true
+
+  validates_presence_of :user
+  validates_presence_of :payment_source
```

Comment:
In Rails 5 `belongs_to` associations are required by default. We don't need these validations.